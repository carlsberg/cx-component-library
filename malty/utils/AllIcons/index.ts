export { allIcons, allIconsStoryOptions } from './AllIconsMap';
