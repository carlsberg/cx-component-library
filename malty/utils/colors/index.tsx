const rgbToHex = (value: string): string =>
  value.replace(/rgb\((\d+),\s*(\d+),\s*(\d+)\)/gi, (_, r, g, b) => {
    const toHex = (n: string) => parseInt(n, 10).toString(16).padStart(2, '0');
    return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
  });

export { rgbToHex };
