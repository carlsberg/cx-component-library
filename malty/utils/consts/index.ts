export enum EventKeys {
  ENTER = 'Enter',
  SPACE = 'Space',
  ESCAPE = 'Escape'
}
