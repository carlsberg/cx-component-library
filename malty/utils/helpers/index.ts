export const getBreakpointNumber = (breakpoint: string) => Number(breakpoint.split('px')[0]);
