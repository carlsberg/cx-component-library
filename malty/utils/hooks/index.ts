export { Device, useMatchMedia } from './useMatchMedia';
