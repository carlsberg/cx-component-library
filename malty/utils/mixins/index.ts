import { css } from 'styled-components';

// Hide an element to all devices except screen readers. More info here: https://webaim.org/techniques/css/invisiblecontent/
export const srOnly = css`
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
`;
