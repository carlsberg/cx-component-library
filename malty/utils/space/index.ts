import type { ArgTypes } from '@storybook/react';

export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}

export const space = <Props extends SpaceProps>({ m = '', mx, my, mt, mr, mb, ml }: Props) => `
    margin: ${m};
    margin-top: ${mt ?? my ?? ''};
    margin-bottom: ${mb ?? my ?? ''};
    margin-right: ${mr ?? mx ?? ''};
    margin-left: ${ml ?? mx ?? ''};
`;

export function spreadSpaceProps<Props extends SpaceProps>(props: Props) {
  const { m, mt, mr, mb, ml, mx, my, ...restProps } = props;

  return { spaceProps: { m, mt, mr, mb, ml, mx, my }, restProps };
}

export function generateStorybookSpacing(): ArgTypes {
  return {
    m: {
      description: 'margin',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mt: {
      description: 'margin-top',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mr: {
      description: 'margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mb: {
      description: 'margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    ml: {
      description: 'margin-left',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mx: {
      description: 'margin-left, margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    my: {
      description: 'margin-top, margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    }
  };
}
