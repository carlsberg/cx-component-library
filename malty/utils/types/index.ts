export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}
