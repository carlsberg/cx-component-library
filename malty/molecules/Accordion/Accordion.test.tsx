import { render } from '@carlsberggbs/malty.utils.test';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Accordion, AccordionItem, AccordionSize } from '.';

jest.mock('uuid', () => ({ v4: () => '00000000-0000-0000-0000-000000000000' }));

describe('Accordion', () => {
  it('should render accordion items', () => {
    render(
      <Accordion size={AccordionSize.Medium}>
        <AccordionItem eventKey="1" title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem eventKey="2" title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );

    expect(screen.getByText('Accordion title 1')).toBeInTheDocument();
    expect(screen.getByText('Accordion title 2')).toBeInTheDocument();
  });

  it('should have the correct test id', () => {
    render(
      <Accordion size={AccordionSize.Medium} dataQaId="accordion">
        <AccordionItem eventKey="1" title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem eventKey="2" title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );

    expect(screen.getByTestId('accordion-accordion-container')).toBeInTheDocument();
  });

  it('should have two accordion items opened at the same time', () => {
    render(
      <Accordion size={AccordionSize.Medium} dataQaId="accordion" alwaysOpen>
        <AccordionItem eventKey="1" title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem eventKey="2" title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );

    const accordionItem1 = screen.getByText('Accordion title 1');
    const accordionItem2 = screen.getByText('Accordion title 2');

    userEvent.click(accordionItem1);
    userEvent.click(accordionItem2);

    const text1 = screen.queryByText('Accordion content 1');
    const text2 = screen.queryByText('Accordion content 2');

    expect(text1).toBeVisible();
    expect(text2).toBeVisible();
  });

  it('should open item content and close the rest when click', () => {
    render(
      <Accordion size={AccordionSize.Medium} dataQaId="accordion">
        <AccordionItem eventKey="1" title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem eventKey="2" title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );

    const accordionItem1 = screen.getByText('Accordion title 1');
    const accordionItem2 = screen.getByText('Accordion title 2');

    const text1 = screen.queryByText('Accordion content 1');
    const text2 = screen.queryByText('Accordion content 2');

    expect(text1).not.toBeVisible();
    expect(text2).not.toBeVisible();

    userEvent.click(accordionItem1);

    expect(text1).toBeVisible();
    expect(text2).not.toBeVisible();

    userEvent.click(accordionItem2);

    expect(text1).not.toBeVisible();
    expect(text2).toBeVisible();
  });

  it('should have the item open according to the defaultActivekey', () => {
    render(
      <Accordion size={AccordionSize.Medium} dataQaId="accordion" defaultActiveKey="1">
        <AccordionItem eventKey="1" title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem eventKey="2" title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );

    const text1 = screen.queryByText('Accordion content 1');
    const text2 = screen.queryByText('Accordion content 2');

    expect(text1).toBeVisible();
    expect(text2).not.toBeVisible();
  });
});
