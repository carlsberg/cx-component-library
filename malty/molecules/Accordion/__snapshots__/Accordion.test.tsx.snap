// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion matches snapshot 1`] = `
<span
  className="sc-bczRLJ hROJNc"
>
  <ul
    className="sc-hKMtZM gGkXTB"
    data-testid="undefined-accordion-container"
  >
    <span
      className="sc-bczRLJ hROJNc"
    >
      <li
        className="sc-eCYdqJ lgICAm accordionItem"
        data-testid="undefined-item"
      >
        <div
          aria-controls="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          aria-expanded={false}
          className="sc-jSMfEi iujwhi accordion-header"
          onClick={[Function]}
          size="32px"
        >
          <span
            className="sc-bczRLJ hROJNc"
          >
            <p
              className="sc-dkzDqf cchzTL"
              color="digital-black"
            >
              Accordion title 1
            </p>
          </span>
          <svg
            className="sc-gsnTZi gIwTyE sc-ftvSup kTJVVU"
            color="#212833"
            data-testid="svg-component"
            size="24px"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              fillRule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 13.586L6.707 8.293a1 1 0 00-1.414 1.414l6 6a1 1 0 001.414 0l6-6a1 1 0 10-1.414-1.414z"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-gKXOVf qEXDZ accordionBody"
          data-testid="undefined-content-container"
          id="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          open={false}
        >
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </li>
    </span>
    <span
      className="sc-bczRLJ hROJNc"
    >
      <li
        className="sc-eCYdqJ lgICAm accordionItem"
        data-testid="undefined-item"
      >
        <div
          aria-controls="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          aria-expanded={false}
          className="sc-jSMfEi iujwhi accordion-header"
          onClick={[Function]}
          size="32px"
        >
          <span
            className="sc-bczRLJ hROJNc"
          >
            <p
              className="sc-dkzDqf cchzTL"
              color="digital-black"
            >
              Accordion title 2
            </p>
          </span>
          <svg
            className="sc-gsnTZi gIwTyE sc-ftvSup kTJVVU"
            color="#212833"
            data-testid="svg-component"
            size="24px"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              fillRule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 13.586L6.707 8.293a1 1 0 00-1.414 1.414l6 6a1 1 0 001.414 0l6-6a1 1 0 10-1.414-1.414z"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-gKXOVf qEXDZ accordionBody"
          data-testid="undefined-content-container"
          id="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          open={false}
        >
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </li>
    </span>
  </ul>
</span>
`;
